plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/AwsApiCallStats.java

```diff
@@ -42,6 +43,12 @@ public <V, E extends Exception> V call(ThrowingCallable<V, E> callable)
         }
     }
 
+    public <V, E extends Exception> ThrowingCallable<V, E> wrapCallableWithTiming(ThrowingCallable<V, E> delegate)
```

Comment:
This is intentionally added, but currently unused within Trino. Internally to Athena, we have places where we wrap certain `Callable<V>` values with an outer callable for reporting timing before passing them into guava caches, etc.

Reply:
Unused helper methods are subject to removal by anyone encountering such an unused method.

Are you suggesting we _should_ be leveraging this for certain operations, but we just don't do so _yet_?

Reply:
If someone comes through and removes this method because they think it’s dead code between now and some point in the future when a usage is added, that’s not a huge deal.

There’s a decent chance that a usage of this method will be added in the future, and having this land as part of the general refactor is going to make our lives easier internally. If you feel strongly that this method should be removed now, I’ll do it- but personally I’d rather leave it in for now unless it’s a blocker for merging this PR.

Reply:
We don't tend to add unused stuff or dead code, especially without understanding how the project and its community benefits from them.

That's why i asked this question: Are you suggesting we should be leveraging this for certain operations, but we just don't do so yet?

Reply:
> Are you suggesting we should be leveraging this for certain operations, but we just don't do so yet?

I'm saying Athena _does_ use this for certain operations, but those operations are not currently present in Trino although they may be upstreamed at some point in the future.

This utility method will be generally useful in any situation where an operation is fed into a `Cache` (and only should be timed if the cache misses and invokes the `Callable<V>` to load it, or when a `Callable<V>` is fed into an `ExecutorService#submit`, so other usages might be added to Trino independently that would benefit from it.

Reply:
Removed this method per @findinpath request.